go/vt/vtadmin/cluster/discovery/discovery_consul.go

```diff
@@ -91,45 +93,69 @@ func NewConsul(cluster *vtadminpb.Cluster, flags *pflag.FlagSet, args []string)
 		"consul service tag identifying -keyspaces_to_watch for vtgates")
 
 	vtgateAddrTmplStr := flags.String("vtgate-addr-tmpl", "{{ .Hostname }}",
-		"Go template string to produce a dialable address from a *vtadminpb.VTGate")
+		"Go template string to produce a dialable address from a *vtadminpb.VTGate "+
+			"NOTE: the .FQDN field will never be set in the addr template context.")
 	vtgateDatacenterTmplStr := flags.String("vtgate-datacenter-tmpl", "",
 		"Go template string to generate the datacenter for vtgate consul queries. "+
 			"The meta information about the cluster is provided to the template via {{ .Cluster }}. "+
 			"Used once during initialization.")
+	vtgateFQDNTmplStr := flags.String("vtgate-fqdn-tmpl", "",
+		"Optional Go template string to produce an FQDN to access the vtgate from a browser. "+
+			"E.g. \"{{ .Hostname }}.example.com\".")
 
 	/* vtctld discovery config options */
 	flags.StringVar(&disco.vtctldService, "vtctld-service-name", "vtctld", "consul service name vtctlds register as")
 
 	vtctldAddrTmplStr := flags.String("vtctld-addr-tmpl", "{{ .Hostname }}",
-		"Go template string to produce a dialable address from a *vtadminpb.Vtctld")
+		"Go template string to produce a dialable address from a *vtadminpb.Vtctld "+
+			"NOTE: the .FQDN field will never be set in the addr template context.")
 	vtctldDatacenterTmplStr := flags.String("vtctld-datacenter-tmpl", "",
 		"Go template string to generate the datacenter for vtgate consul queries. "+
 			"The cluster name is provided to the template via {{ .Cluster }}. "+
 			"Used once during initialization.")
+	vtctldFQDNTmplStr := flags.String("vtctld-fqdn-tmpl", "",
+		"Optional Go template string to produce an FQDN to access the vtctld from a browser. "+
+			"E.g. \"{{ .Hostname }}.example.com\".")
 
 	if err := flags.Parse(args); err != nil {
 		return nil, err
 	}
 
+	/* gates options */
 	if *vtgateDatacenterTmplStr != "" {
 		disco.vtgateDatacenter, err = generateConsulDatacenter("vtgate", cluster, *vtgateDatacenterTmplStr)
 		if err != nil {
 			return nil, fmt.Errorf("failed to generate vtgate consul datacenter from template: %w", err)
 		}
 	}
 
+	if *vtgateFQDNTmplStr != "" {
+		disco.vtgateFQDNTmpl, err = template.New("consul-vtgate-fqdn-template-" + cluster.Id).Parse(*vtgateFQDNTmplStr)
+		if err != nil {
+			return nil, fmt.Errorf("failde to parse vtgate FQDN template %s: %w", *vtgateFQDNTmplStr, err)
```

Comment:
Typo: "failde" (there's another instance lower in the file)

Reply:
it's the Olde English of "failed" (thank you!)